### ISSUE TITLE
Overcoming Catastrophic Forgetting in Incremental Few-Shot Learning by Finding Flat Minima

### ISSUE THREAD

**Issue by chaerinkong** [reyllama/MIPAL-CL#3]
## TL; DR

- SOTA IFL methods lag behind a simple baseline of *frozen representation + NN classifier*.
- Optimize with model parameter noise perturbation to find flat loss minima (robustness-to-perturbation sense)
- Clamp parameter updates with heuristic (hyperparam) to ensure changes within flat region.

## Motivation

- IFL being such a challenging task, it's mostly better to ***not learn anything*** from ensuing tasks.

### G-Dumb-ish Observation

![image](https://user-images.githubusercontent.com/46308407/158131295-4cea7208-f4aa-4829-be95-043a70867c51.png)

### Desideratum

![image](https://user-images.githubusercontent.com/46308407/158131327-1196648d-3c43-46c4-9678-0ce41d36eb98.png)

### Objective

![image](https://user-images.githubusercontent.com/46308407/158131396-4e2fd370-3605-414b-9732-20bcf2216dac.png)

### Algorithm

![image](https://user-images.githubusercontent.com/46308407/158131436-2138eccc-674a-41de-baf6-715a686e8cc2.png)

## Evaluation

![image](https://user-images.githubusercontent.com/46308407/158131486-51d7b7be-f34b-45b6-a8d9-6b80888b0355.png)
![image](https://user-images.githubusercontent.com/46308407/158131502-f14b0ff0-1e65-42c4-aa01-8288c3f6a5ea.png)
![image](https://user-images.githubusercontent.com/46308407/158131521-041f8818-37a0-44ec-a67d-9a435c01b373.png)

## Comments

- The initial observation is insightful, while the proposed method is not particularly appealing (extremely passive approach)
